pkg/shell/shell.scss

```diff
@@ -141,17 +141,25 @@ iframe.container-frame {
 }
 
 #credentials-dialog {
-    div.table-scrollable {
-        max-height: 500px;
-        overflow-y: auto;
-        margin-right: 7px;
-        padding-right: 8px;
-        margin-top: 15px;
-        padding-top: 0px;
+    .pf-c-modal-box__body {
+        // Turn off modal scrolling
+        overflow: visible;
+    }
+
+    .modal-body {
+        display: grid;
+        grid-template-rows: auto 1fr;
+        // Full screen minus the button, heading, and padding
+        max-height: calc(100vh - (var(--pf-global--spacer--md) * 11));
```

Comment:
This is still a "magic number" in a way, but it's based on the standard PatternFly unit and only applies to the heading, button height, and the padding. All of the content is handled inside of this specific grid and the sizing of the list wrapper (table-scrollable) is whatever is left over after the top of the dialog.

(Previous iteration used a magic number that also guesstimated the text at the top and sizing was done in the wrapper. This is much better. Both are better than what we have in Cockpit currently, though.)